src/js/components/Analytics.jsx

```diff
@@ -95,8 +95,16 @@ class Analytics extends Component {
     dispatch(fetchAnalytics(source))
   }
 
-  onChangeValue (index, value) {
+  onChangeValue (index, value, field, data) {
     const {dispatch} = this.props
+    
+    if (data) {
```

Comment:
Here is how I would write this:

```javascript
onChangeValue (index, value, field, fields) { // <-- note the change from data to fields. fields is more descriptive
  const {dispatch} = this.props
    
  fields.forEach((item) => {
    const isCurrentField = field === item.name
    const fieldIsString = item.datatype === 'String'
    const shouldConvert = isCurrentField && !fieldIsString
    
    if (!shouldConvert) {
      return
    }
    
    value = JSON.parse(value) // <-- This can get us Numbers and Booleans and null
  })

  dispatch(editFilter(index, {value}))
}
```